.github/workflows/test.yml

```diff
@@ -30,6 +30,30 @@ jobs:
           COUCHDB_PASSWORD: 'test'
           COUCHDB_NODENAME: '127.0.0.1'
           LUCENE_PORT: '5985'
-          SECONDS_TO_WAIT: '10'
+          SECONDS_TO_WAIT: '30'
+          MCM_WARMUP_SECONDS: '10'
+          MCM_COUCHDB_URL: 'http://couchdb:5984/'
+          COUCH_CRED: 'Basic dGVzdDp0ZXN0'
+          MCM_LUCENE_URL: 'http://couchdb-lucene:5985/'
+          MCM_EXAMPLE_DATA_URL: ${{ secrets.MCM_EXAMPLE_DATA_URL }}
+
+        run: |
+          bash deploy/action.sh
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11.x'
+
+      - name: Install test dependencies
+        run: |
+          export REPO_PATH=$GITHUB_WORKSPACE/repo/mcm
+          export TEST_FOLDER_PATH=$REPO_PATH/tests
+          python3 -m pip install --upgrade pip
+          python3 -m pip install -r $TEST_FOLDER_PATH/requirements.txt
+
+      - name: Run test suite
         run: |
-          bash deploy/action.sh
\ No newline at end of file
+          export REPO_PATH=$GITHUB_WORKSPACE/repo/mcm
+          export PYTHONPATH=$REPO_PATH
```

Comment:
Side note: we'll want to create a proper package to avoid this hack (poetry?).

Reply:
Maybe in the future, we could split different parts of the web application as simple independent Python modules. For the moment, I don’t see the need to use something like `poetry` to package the application.